service/src/main/java/com/ge/predix/acs/service/policy/evaluation/SubjectAttributeResolver.java

```diff
@@ -0,0 +1,56 @@
+package com.ge.predix.acs.service.policy.evaluation;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.ge.predix.acs.model.Attribute;
+import com.ge.predix.acs.privilege.management.PrivilegeManagementService;
+import com.ge.predix.acs.rest.BaseSubject;
+
+public class SubjectAttributeResolver {
+
+    private final Map<String, Set<Attribute>> subjectAttributeMap = new HashMap<>();
+    private final PrivilegeManagementService privilegeService;
+    private final String subjectIdentifier;
+    private final Set<Attribute> supplementalSubjectAttributes;
+
+    public SubjectAttributeResolver(final PrivilegeManagementService privilegeService, final String subjectIdentifier,
+            final Set<Attribute> supplementalSubjectAttributes) {
+        this.privilegeService = privilegeService;
+        this.subjectIdentifier = subjectIdentifier;
+        if (null == supplementalSubjectAttributes) {
+            this.supplementalSubjectAttributes = Collections.emptySet();
+        } else {
+            this.supplementalSubjectAttributes = supplementalSubjectAttributes;
+        }
+    }
+
+    public SubjectAttributeResolverResult getResult(final String resourceIdentifier, final Set<Attribute> scopes) {
```

Comment:
return Set<Attribute> instead of SubjectAttributeResolverResult, and remove SubjectAttributeResolverResult